### PR TITLE
SRVLOGIC-184: Review documentation for TP2

### PR DESCRIPTION
**Pull request by domhanak**
JIRA: [SRVLOGIC-184](https://issues.redhat.com/browse/SRVLOGIC-184)

@ricardozanini @krisv Please take a look, we are missing the guides from https://github.com/apache/incubator-kie-kogito-docs/pull/519 I will add them once they are merged today.

Updates everything to be TP2 ready.

Cc: @mgencur

### CODE REVIEW

File: modules/serverless-logic/pages/persistence/postgresql-flyway-migration.adoc [hunk 1]
```diff
@@ -13,7 +14,7 @@ When you upgrade your {product_name} version, by default it **won't** pick up th
 
 === Migrate using Flyway Config
 
-* {product_name} provides a mechanism for migrating your database while updating the {product_name} version using the following `Flyway` properties (default value is `false`, not upgrade):
+* Kogito provides a mechanism for migrating your database while updating the Kogito version using the following `Flyway` properties (default value is `false`, not upgrade):
```
Comment (ricardozanini) [inline modules/serverless-logic/pages/persistence/postgresql-flyway-migration.adoc:17]: Maybe we need to revert to `{product_name}`

File: modules/serverless-logic/pages/core/understanding-jq-expressions.adoc [hunk 2]
```diff
@@ -1,7 +1,7 @@
-= jq expressions in {product_name}
+= jq Expressions
```
[the remark this answers is not in the document]
Comment (domhanak) [reply modules/serverless-logic/pages/core/understanding-jq-expressions.adoc:1]: Add it, also approved the upstream.

**Comment by domhanak** [comment]
Oh so it seems the rename missed some spots, thanks @ricardozanini fixing it

File: modules/serverless-logic/pages/core/custom-functions-support.adoc [hunk 3]
```diff
@@ -599,7 +576,7 @@ Using the recent approach, you can write a link:{kogito_sw_examples_url}/serverl
 
 The `operation` starts with `rpc`, which is the custom type identifier, and continues with `division`, which denotes the operation that will be executed in the legacy server.
 
-A Kogito addon that defines the `rpc` custom type must be developed for this function definition to be identified. It is consist of a link:{kogito_sw_examples_url}/serverless-workflow-custom-type/serverless-workflow-custom-rpc-deployment[deployment project] and a link:{kogito_sw_examples_url}/serverless-workflow-custom-type/serverless-workflow-custom-rpc[runtime project].
+A { product_name } addon that defines the `rpc` custom type must be developed for this function definition to be identified. It is consist of a link:{kogito_sw_examples_url}/serverless-workflow-custom-type/serverless-workflow-custom-rpc-deployment[deployment project] and a link:{kogito_sw_examples_url}/serverless-workflow-custom-type/serverless-workflow-custom-rpc[runtime project].
```
Comment (ricardozanini) [inline modules/serverless-logic/pages/core/custom-functions-support.adoc:579]: I think these spaces might cause an issue during parse, no?

**Review by krisv — APPROVED** [review]

**Pull request merged**